scripts/fix_security_group.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+import gzip
+import json
+import logging
+import os
+import platform
+import urllib.parse
+import urllib.request
+import urllib.response
+
+
+def get_http_response_str(response: urllib.response.addinfourl) -> str:
+    if response.info().get("Content-Encoding") == "gzip":
+        pagedata = gzip.decompress(response.read())
+    elif response.info().get("Content-Encoding") == "deflate":
+        pagedata = response.read()
+    else:
+        pagedata = response.read()
+
+    return pagedata
+
+
+def get_bearer_token(
+    refresh_token: str, cloud_base_domain: str = "cloud.ibm.com"
+) -> str:
+    url = f"https://iam.{cloud_base_domain}/identity/token"
+
+    payload = {
+        "grant_type": "refresh_token",
+        "refresh_token": refresh_token,
+        "client_id": "bx",
+        "client_secret": "bx",  # pragma: allowlist secret
+    }
+
+    headers = {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "User-Agent": f"python-{platform.python_version()}",
+        "Accept-Encoding": "gzip, deflate",
+        "Accept": "*/*",
+        "Connection": "keep-alive",
+    }
+
+    data = urllib.parse.urlencode(payload)
+    data = data.encode("ascii")
+    request = urllib.request.Request(url=url, data=data, headers=headers)
+
+    response = urllib.request.urlopen(request)
+
+    if response.status == 200:
+        response_txt = get_http_response_str(response=response)
+        return json.loads(response_txt).get("access_token")
+    else:
+        raise Exception("Failed to retrieve refresh_token")
+
+
+def get_security_group_rules(
+    sg_id: str, region: str, access_token: str, cloud_base_domain: str = "cloud.ibm.com"
+) -> dict:
+    url = f"https://{region}.iaas.{cloud_base_domain}/v1/security_groups/{sg_id}/rules?version=2023-04-11&generation=2"
```

Comment:
Going to break for VPC VPE. Eg: the domain is eu-de.private.iaas.cloud.ibm.com for the VPC VPE in Frankfurt